src/extension/op_def.rs

```diff
@@ -326,36 +343,40 @@ impl Extension {
         &mut self,
         name: SmolStr,
         description: String,
-        params: Vec<TypeParam>,
+        params: (Vec<TypeParam>, Vec<TypeParam>),
```

Comment:
Why a tuple of param vectors instead of two arguments?

Reply:
`clippy: too many arguments`....

But yeah, better to disable that for this function, and add a TODO to sort this out when we've converted existing binary cases to type schemes.

@ss2165 if we went with "declare TypeParams to binary function only" (rather than declaring in advance the params that will be taken by the type scheme that the binary function gives you back) then that'd solve this. Of course this is not a reason to do that....